<robot name="quad_robot">
  <ros2_control name="VescSystem" type="system">
    <hardware>
      <plugin>vesc_nexus/VescSystemHardwareInterface</plugin>
      <param name="can_interface">can0</param>
      <param name="publish_rate">50.0</param>
      <param name="wheel_radius">0.115</param>
    </hardware>

    <joint name="front_left_wheel" type="continuous">
      <command_interface name="velocity"/>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
      <param name="can_id">49</param>
      <param name="poles">30</param>
      <param name="min_erpm">100</param>
      <param name="wheel_radius">0.115</param>
    </joint>

    <joint name="front_right_wheel" type="continuous">
      <command_interface name="velocity"/>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
      <param name="can_id">124</param>
      <param name="poles">30</param>
      <param name="min_erpm">100</param>
      <param name="wheel_radius">0.115</param>
    </joint>

    <joint name="rear_left_wheel" type="continuous">
      <command_interface name="velocity"/>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
      <param name="can_id">81</param>
      <param name="poles">30</param>
      <param name="min_erpm">100</param>
      <param name="wheel_radius">0.115</param>
    </joint>

    <joint name="rear_right_wheel" type="continuous">
      <command_interface name="velocity"/>
      <state_interface name="velocity"/>
      <state_interface name="position"/>
      <param name="can_id">94</param>
      <param name="poles">30</param>
      <param name="min_erpm">100</param>
      <param name="wheel_radius">0.115</param>
    </joint>
  </ros2_control>
</robot>